src/Storages/IRemoteFileMetadata.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include <memory>
+#include <base/types.h>
+#include <functional>
+#include <unordered_map>
+#include <boost/core/noncopyable.hpp>
+
+namespace DB
+{
+class IRemoteFileMetadata
+{
+public:
+    IRemoteFileMetadata() = default;
```

Comment:
I remember I might have asked this before, but still is it possible to get rid of default constructor? I think it should be possible.

Reply:
It would not be passed in compiling
```
error: constructor for 'DB::StorageHiveMetadata' must explicitly initialize the base class 'DB::IRemoteFileMetadata' which does not have a default constructor
```

Reply:
done